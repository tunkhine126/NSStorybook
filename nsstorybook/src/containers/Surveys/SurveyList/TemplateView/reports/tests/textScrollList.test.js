/**
 *
 * Tests for TextScrollList
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import TestingComponent from 'components/shared/Testing';
import { render } from '@testing-library/react';

import TextScrollList from '../textScrollList';

describe('<TextScrollList />', () => {
  const props = {
    responses: [],
    primaryRecipientIdPath: '/recipients/family-list/family/:id',
    surveyName: 'Test Survey',
  };

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <TestingComponent>
        <TextScrollList {...props} />
      </TestingComponent>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <TestingComponent>
        <TextScrollList {...props} />
      </TestingComponent>
    );
    expect(firstChild).toMatchSnapshot();
  });
});
