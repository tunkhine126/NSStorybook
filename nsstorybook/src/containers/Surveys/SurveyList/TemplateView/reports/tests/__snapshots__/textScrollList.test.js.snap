// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextScrollList /> Should render and match the snapshot 1`] = `
<div
  style="position: relative; height: 241px; width: 90%; overflow: auto; will-change: transform; direction: ltr; margin-left: 40px;"
>
  <div
    style="height: 0px; width: 100%;"
  />
</div>
`;
